backend/app/resources.py

```diff
@@ -0,0 +1,111 @@
+from flask_restful import Resource, fields, marshal_with, marshal
+from sqlalchemy import exc
+
+from app.marshallers import item_marshaller, category_marshaller, subcategory_marshaller
+from app.models import Item, Category, Subcategory
+from app.parsers import item_parser, subcategory_parser, category_parser, creating_item_parser, \
+    creating_category_parser, creating_subcategory_parser
+from init_app import db
+
+
+class Items(Resource):
+    @marshal_with(item_marshaller)
+    def get(self, item_id):
+        return Item.query.get_or_404(item_id)
+
+    @marshal_with(item_marshaller)
+    def delete(self, item_id):
+        item = Item.query.get_or_404(item_id)
+        db.session.delete(item)
+        db.session.commit()
+        return '', 204
+
+    @marshal_with(item_marshaller)
+    def put(self, item_id):
+        item = Item.query.get_or_404(item_id)
+        args = item_parser.parse_args()
+        item.update_item(args)
+        return item, 201
+
+
+class ItemsList(Resource):
+    def get(self):
+        return marshal(Item.query.all(), item_marshaller)
+
+    def post(self):
+        args = creating_item_parser.parse_args()
+        item = Item(name=args['name'])
+        try:
+            db.session.add(item)
+            db.session.commit()
+            item.update_item(args)
+        except exc.IntegrityError:
+            db.session.rollback()
+            return 404
+        return marshal(item, item_marshaller), 201
+
+
+class Categories(Resource):
+    def get(self, category_id):
+        return marshal(Category.query.get_or_404(category_id), category_marshaller)
+
+    def delete(self, category_id):
+        item = Category.query.get_or_404(category_id)
+        db.session.delete(item)
+        db.session.commit()
+        return marshal('', category_marshaller), 204
+
+    def put(self, category_id):
+        category = Category.query.get_or_404(category_id)
+        args = category_parser.parse_args()
+        category.update_category(args)
+        return marshal(category, category_marshaller), 201
+
+
+class CategoryList(Resource):
+    def get(self):
+        return marshal(Category.query.all(), category_marshaller)
+
+    def post(self):
+        args = creating_category_parser.parse_args()
+        category = Category(name=args['name'])
+        try:
+            db.session.add(category)
```

Comment:
Probably you could move this `try/except` block to separate method? Because it occurs a couple times in the code.